src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/ConstantBuilder.java

```diff
@@ -340,9 +329,9 @@ private Constant emitConstantSegment(String javaName, Object value) {
         append(memberMods());
         append("MemorySegment ");
         append(fieldName);
-        append(" = CLinker.toCString(\"");
+        append(" = ResourceScope.newSharedScope().allocateUtf8String(\"");
```

Comment:
Do we need shared scope here? or confined scope will do?

Reply:
This is a string constant - so we don't know which thread will use it. It cannot be confined.

Reply:
got it